### DISCUSSION TITLE
Please help me fix my program - Guess the Number(Computer)

### DISCUSSION THREAD

**Discussion by SimonHuangxp** [community/community#115448] [Programming Help]
### Select Topic Area

Question

### Body

Hi, so I made a project called guess the number, here's how it works. So the player will enter a number between 1 and 100, and the computer will try to guess the number under 10 tries. If the computer ran out of tries, you win. Otherwise, the computer wins. I will be using binary search to guess the number. But the math equation I used for it doesn't quite work, sometimes it will check for the wrong thing, and guess for the wrong number I want it to guess. I am trying to find a equation that works for all the conditions, but I could not find one. So if you can help me find a equation that works or fix a few bugs or even improve this program, please do so.

here's a link to my project:
https://replit.com/@BruceDoesCoding/Guess-the-Number-V2

*I am new to programming and the code may look not as good

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @SimonHuangxp ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions. 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by Manjunathhs-0003** [comment]
def guessing():
    # Game Variables
    win = False
    tries = 10
    low = 1
    high = 100

    print("Welcome to the game of guess the number.")
    print("In this game,")
    print("You will think of a number between 1-100,")
    print("And the computer is gonna try to guess it.")
    print("Good Luck!")

    UserNum = int(input("Please enter your number: "))

    # Guessing Loop
    while tries > 0:
        if win:
            break

        GuessNum = (low + high) // 2
        print(f"The computer guesses: {GuessNum}")

        response = input("Is the number correct? (yes/smaller/bigger): ").lower()

        if response == "yes":
            print("Hooray! The computer won!")
            win = True
        elif response == "smaller":
            high = GuessNum - 1
            print(f"Computer has {tries - 1} tries left.")
        elif response == "bigger":
            low = GuessNum + 1
            print(f"Computer has {tries - 1} tries left.")
        else:
            print("Invalid response. Please enter 'yes', 'smaller', or 'bigger'.")

        tries -= 1

    if not win:
        print("Congratulations! You win!")

# Start the game
guessing()

**Reply by Xhst** [reply]
tries should not be decremented in case of invalid input

**Comment by Xhst** [comment]
After the second guess the code keeps guessing the same number.
You should also handle some extra scases (invalid input, out of bounds etc)
Here's a way to fix the code:
```
def guessing():
    # Game Variables
    win = False
    tries = 10
    lower_bound = 1
    upper_bound = 100
    GuessNum = 50

    print("Welcome to the game of guess the number.")
    print(f"In this game, you think of a number between {lower_bound}-{upper_bound} and the computer is going to try to guess it in {tries} tries or less.")
    print("Good Luck!")

    UserNum = int(input("Please enter your number: "))

    # Guessing Loop
    while tries > 0:
        if win:
            break

        guess = input(f"Is the number {GuessNum}? (yes/smaller/bigger): ")

        if guess.lower() == "yes":
            print("Hooray! The computer won!")
            win = True
        elif guess.lower() == "smaller":
            upper_bound = GuessNum - 1
            GuessNum = (lower_bound + upper_bound) // 2
            tries -= 1
            print(f"Computer has {tries} tries left.")
        elif guess.lower() == "bigger":
            lower_bound = GuessNum + 1
            GuessNum = (lower_bound + upper_bound) // 2
            tries -= 1
            print(f"Computer has {tries} tries left.")
        else:
            print("Invalid input. Please enter 'yes', 'smaller', or 'bigger'.")

    if not win:
        print("Congratulations! You win!")

guessing()
```
Other considerations:
There's no need to ask the user to input the number (you're not using the variable UserNum)
I modified the print to have it correct even with different boundaries

**Comment by Manjunathhs-0003** [comment]
def guessing():
    # Game Variables
    win = False
    tries = 10
    lower_bound = 1
    upper_bound = 100
    GuessNum = (lower_bound + upper_bound) // 2

    print("Welcome to the game of guess the number.")
    print(f"In this game, the computer will try to guess the number you're thinking of between {lower_bound}-{upper_bound} in {tries} tries or less.")
    print("Good Luck!")

    # Guessing Loop
    while tries > 0:
        if win:
            break

        guess = input(f"Is the number {GuessNum}? (yes/smaller/bigger): ")

        if guess.lower() == "yes":
            print("Hooray! The computer won!")
            win = True
        elif guess.lower() == "smaller":
            if GuessNum == lower_bound:  # Check if already at lower bound
                print("The number cannot be smaller than the lower bound.")
                continue  # Skip decrementing tries
            upper_bound = GuessNum - 1
            GuessNum = (lower_bound + upper_bound) // 2
            print(f"Computer has {tries - 1} tries left.")
        elif guess.lower() == "bigger":
            if GuessNum == upper_bound:  # Check if already at upper bound
                print("The number cannot be bigger than the upper bound.")
                continue  # Skip decrementing tries
            lower_bound = GuessNum + 1
            GuessNum = (lower_bound + upper_bound) // 2
            print(f"Computer has {tries - 1} tries left.")
        else:
            print("Invalid input. Please enter 'yes', 'smaller', or 'bigger'.")
            continue  # Skip decrementing tries for invalid input

        tries -= 1

    if not win:
        print("Congratulations! You win!")

guessing()

**Comment by SimonHuangxp** [comment]
Thanks a lot to all of you who responded! It certainly helped a lot,  very appreciated!
